lib/Driver/Driver.cpp

```diff
@@ -1788,7 +1798,7 @@ Driver::computeCompilerMode(const DerivedArgList &Args,
     }
     if (ArgRequiringSinglePrimaryCompile)
       Diags.diagnose(SourceLoc(), diag::warn_ignoring_source_range_dependencies,
-                     ArgRequiringSingleCompile->getOption().getPrefixedName());
+                     ArgRequiringSinglePrimaryCompile->getOption().getPrefixedName());
```

Comment:
Was this a bug in the earlier code/why did this part change?

Reply:
Ah yes. Removed it, I think this was probably the original intent, and I might submit a different PR containing it to see, do you know?

Reply:
Looking at the code, I think the existing code is correct (but we don't seem to have a test for it). What that surrounding branch (this one and the one before it) is saying is that if we are given `-wmo` or `-index-file`, that takes precedence over both `-batch-mode` and `-enable-source-range-dependencies` (maybe those aren't the exact flag names but you get the idea).  So we emit a diagnostic saying "hey, this -wmo/-index-file flag is taking precedence so I am ignoring this thing (listed in the diag::blah)". That flag is stored in the `ArgRequiringSingleCompile` variable. But yeah, it is certainly somewhat confusing.

Reply:
So since you've undone the change in the second commit, it should be fine. :+1: